packages/l/libsdl_image/xmake.lua

```diff
@@ -50,23 +52,17 @@ package("libsdl_image")
     end)
 
     on_install("macosx", "linux", function (package)
-        local exflags
-        local configs = {"--disable-dependency-tracking"}
+        local configs = {}
         table.insert(configs, "--enable-shared=" .. (package:config("shared") and "yes" or "no"))
+        table.insert(configs, "--enable-static=" .. (package:config("shared") and "no" or "yes"))
+        if package:is_plat("linux") and package:config("pic") ~= false then
+            table.insert(configs, "--with-pic")
+        end
         local libsdl = package:dep("libsdl")
         if libsdl and not libsdl:is_system() then
             table.insert(configs, "--with-sdl-prefix=" .. libsdl:installdir())
         end
-        if package:version():eq("2.0.5") then
-            io.replace("Makefile.am", "noinst_PROGRAMS = showimage", "")
```

Comment:
We need to be compatible with older versions, 2.0.5 need this patch

Reply:
how do i tell the test script to specifically install the 2.0.5 version to test it?

Reply:
test.lua "libsdl_image 2.0.5"

Reply:
so shallow and non-shallow both work at 2.0.5, but i can add it back

Reply:
It will fix `link showanim binary error`, it still does not work on macos ci now.

Maybe we need also disable it in new version.

Reply:
i'm kinda stumped why it fails with `configure: error: C compiler cannot create executables` for linux, or `.libs/libSDL2_image.a(IMG_tga.o) has no symbols` for mac

Reply:
I fixed it, but has an anther error on macOS.


```lua
Undefined symbols for architecture x86_64:
  "_IMG_Init", referenced from:
      _main in _C2F4F90BE61F471084D12FE0A7112520.o
```